File: backend/app/Models/Voucher.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Voucher extends Model
{
    protected $fillable = [
        'name',
        'code',
        'type',
        'value',
        'start_date',
        'expired_date',
        'min_purchase',
        'quota'
    ];

    protected $casts = [
        'value' => 'float',
        'min_purchase' => 'float',
        'start_date' => 'datetime',
        'expired_date' => 'datetime',
    ];

    public function userVouchers()
    {
        return $this->hasMany(UserVoucher::class);
    }
}

File: frontend/lib/models/voucher.dart
class Voucher {
  final int id;
  final String name;
  final String? code;
  final String type;
  final double value;
  final double minPurchase;
  final String? expiredDate;

  Voucher({
    required this.id,
    required this.name,
    this.code,
    required this.type,
    required this.value,
    required this.minPurchase,
    this.expiredDate,
  });

  factory Voucher.fromJson(Map<String, dynamic> json) {
    return Voucher(
      id: json['id'],
      name: json['name'],
      code: json['code'],
      type: json['type'],
      value: double.parse(json['value'].toString()),
      minPurchase: double.parse(json['min_purchase'].toString()),
      expiredDate: json['expired_date'],
    );
  }
}
